docs/microservices/problems.md

```diff
@@ -2,292 +2,255 @@
+| 422 | Missing or invalid problemSetId or body |
+| 500 | Internal server error |
+
+<br>
+
+> DELETE /problemSets/{problemSetId}
+
+**Params**
 
-Response:
-```json
+* `problemSetId`: ID for problemSet to delete
+
+**Responses**
+
+| Status | Response |
+| --- | --- |
+| 200 | Empty |
+| 404 | ProblemSet not found |
+| 422 | Missing or invalid problemSetId |
+| 500 | Internal server error |
+
+<br>
+
+### Problems
+
+<br>
+
+> Model
+
+```javascript
 {
-    "problemMetadata": {
-    "platformProblemId": "1352G",
-    "difficulty": "easy"
-    },
-    "problemSetIds": [
-    "5eb855b2cf015c0062d71f88"
-    ],
-    "_id": "5eb8570217271500be61b2c9",
-    "title": "Test Problem 1: Special Permutation",
-    "source": "CODEFORCES",
-    "sourceLink": "https://codeforces.com/problemset/problem/1352/G",
-    "problemId": "6e88ec57160cd8164d9460a9885fbece7047c13c",
-    "createdAt": "2020-05-10T19:33:22.888Z",
-    "updatedAt": "2020-05-10T19:33:22.888Z",
-    "__v": 0
+    title: String,
+    source: String,
+    problemId: String,
+    sourceLink: String,
+    problemSetIds: [String],
+    problemMetadata: {
+        platformProblemId: String,
+        difficulty: String
+    }
 }
 ```
+
+NOTE: `problemId` is generated by the backend, and should not be included in POST/PUT requests
+
+<br>
+
+> GET /problems
+
+**Responses**
+
+| Status | Response |
+| --- | --- |
+| 200 | Array of problems |
+| 500 | Internal server error |
+
+<br>
+
+> GET /problems/{problemId}
+
+**Params**
+
+* `problemId`: ID for problem to fetch
+
+**Responses**
+
+| Status | Response |
+| --- | --- |
+| 200 | Problem |
+| 404 | Problem not found |
+| 422 | Missing or invalid problemId |
+| 500 | Internal server error |
+
+<br>
+
+> POST /problems
+
+**Body**
+
+See Problem model, but don't include `problemId` parameter in the body
+
+**Responses**
+
+| Status | Response |
+| --- | --- |
+| 200 | Problem |
+| 422 | Missing or invalid params in body |
+| 500 | Internal server error |
+
+<br>
+
+> PUT /problems/{problemId}
+
+**Params**
+
+* `problemId`: ID for problem to update
+
+**Body**
+
+See Problem model, but don't include `problemId` parameter in the body
+
+**Responses**
+
+| Status | Response |
+| --- | --- |
+| 200 | Problem |
+| 404 | Problem not found |
+| 422 | Missing or invalid problemId or body |
+| 500 | Internal server error |
+
+<br>
+
+> DELETE /problems/{problemId}
+
+**Params**
+
+* `problemId`: ID for problem to delete
+
+**Responses**
+
+| Status | Response |
+| --- | --- |
+| 200 | Empty |
+| 404 | Problem not found |
+| 422 | Missing or invalid problemSetId |
+| 500 | Internal server error |
+
+<br>
+
+> GET /problems/{problemId}/exists
+
+**Params**
+
+* `problemId`: ID for problem to delete
+    * This `problemId` that is used in this requests URL is a custom ID that can be generated by concatenating the platform ("CODEFORCES") and the problem ID as displayed on the platform ("1352G"), and then taking the SHA-1 hash of the resulting string.
```

Comment:
shouldn't specify the way we generate codeforces ids as it can be differenct for other platforms